### PR TITLE
Top level epa response

### PR DESCRIPTION
**Pull request by tbock**

### CODE REVIEW

File: QuestionLibrary/admin.py [hunk 1]
```diff
@@ -59,6 +62,103 @@ def load_selected_records_action(modeladmin, request, queryset):
...
+    search_fields = ['name']
+    form = EPAResponseForm
+
+    def _status(self, obj=None):
+        if obj.disabled_date is None:
+            return 'Active'
+        else:
+            return f'Disabled {obj.disabled_date.strftime("%Y-%m-%d")}'
+
+    def _map_service_config(self, obj=None):
```
Comment (tbock) [inline QuestionLibrary/admin.py:133]: move to the model

File: QuestionLibrary/admin.py [hunk 2]
```diff
@@ -59,6 +62,103 @@ def load_selected_records_action(modeladmin, request, queryset):
...
+                return 'Error getting configuration'
+            if 'layers' in config:
+                return 'Configuration Downloaded'
+
+    def save_model(self, request, obj, form, change):
+        if 'map_service_url' in form.changed_data:
+            obj.get_map_service(request.user)
+        super().save_model(request, obj, form, change)
+
+    def get_search_results(self, request, queryset, search_term):
```
Comment (tbock) [inline QuestionLibrary/admin.py:148]: If not dealing fk/relationships I think you can use limit_choices_to on the model fk field.

File: QuestionLibrary/admin.py [hunk 3]
```diff
@@ -203,33 +301,27 @@ class RelatedQuestionListAdmin(admin.ModelAdmin):
...
-        if self.instance.service_config is not None:
-            config = json.loads(self.instance.service_config)
-
-            if type(config) is dict:
-                layer_choice = [(x['id'], x['name']) for x in config['layers']]
-                self.fields['layer'].choices = layer_choice
-
-                layer_choice = [(x['id'], x['name']) for x in config['tables']]
-                self.fields['assessment_layer'].choices = layer_choice
+        if self.instance.epa_response and self.instance.epa_response.map_service_config is not None:
```
Comment (tbock) [inline QuestionLibrary/admin.py:322]: Please add a comment to clarify what this is doing.

**Review by tbock — COMMENTED** [review]
Thanks.

**Comment by rdmccann** [comment]
ready for review after addressing comments

**Pull request merged**